src/Analyzer.JsonRuleEngine/Operators/InequalityOperator.cs

```diff
@@ -0,0 +1,127 @@
+    /// </summary>
+    internal class InequalityOperator : LeafExpressionOperator
+    {
+        /// <inheritdoc/>
+        public override string Name => GetName();
+
+        /// <summary>
+        /// Whether the operator also considers equality
+        /// </summary>
+        public Boolean OrEquals;
+
+        /// <summary>
+        /// Creates an InequalityOperator.
+        /// </summary>
+        /// <param name="specifiedValue">The value specified in the JSON rule.</param>
+        /// <param name="isNegative">Whether the operator compares by greater than or by less than.</param>
+        /// <param name="orEquals">Whether the operator also considers equality.</param>
+        public InequalityOperator(JToken specifiedValue, bool isNegative, bool orEquals)
+        {
+            if (specifiedValue == null)
+            {
+                throw new ArgumentNullException(nameof(specifiedValue));
+            }
+            else
+            {
+                ValidateComparisonTerm(specifiedValue);
+            }
+
+            this.SpecifiedValue = specifiedValue;
+            this.IsNegative = isNegative;
+            this.OrEquals = orEquals;
+        }
+
+        /// <summary>
+        /// Evaluates <paramref name="tokenToEvaluate"/> to determine if it is greater, greaterOrEquals, less, or lessOrEquals than the specified value.
+        /// </summary>
+        /// <param name="tokenToEvaluate">The JToken to evaluate.</param>
+        /// <returns>A value indicating whether or not the evaluation passed.</returns>
+        public override bool EvaluateExpression(JToken tokenToEvaluate)
+        {
+            if (tokenToEvaluate == null)
+            {
+                // If the specified property in the JSON is not defined then we would assume it could potentially have an undesired value
+                return false;
+            }
+
+            ValidateComparisonTerm(tokenToEvaluate);
+
+            // TODO throw this exception earlier?
+            if ((SpecifiedValue.Type == JTokenType.Date && tokenToEvaluate.Type != JTokenType.Date) ||
+                (tokenToEvaluate.Type == JTokenType.Date && SpecifiedValue.Type != JTokenType.Date))
+            {
+                throw new InvalidOperationException($"Cannot compare {SpecifiedValue.Type} with {tokenToEvaluate.Type} using an InequalityOperator");
+            }
+
+            var normalizedSpecifiedValue = GetNormalizedValue(SpecifiedValue);
+            var normalizedTokenToEvaluate = GetNormalizedValue(tokenToEvaluate);
+
+            var result = IsNegative ? normalizedSpecifiedValue < normalizedTokenToEvaluate : normalizedSpecifiedValue > normalizedTokenToEvaluate;
+
+            if (OrEquals)
+            {
+                result = result || normalizedSpecifiedValue == normalizedTokenToEvaluate;
+
+            }
+
+            return result;
+        }
+
+        // TODO throw this exception earlier?
+        private void ValidateComparisonTerm(JToken term)
```

Comment:
I think as long as the validation happens during the "rule interpretation" phase and before evaluating a template, it should be ok.  This would fall in the "interpretation" phase (creating the rule) so I think it's fine.